### PR TITLE
TT-382: generate .yml files and post to zendesk

### PR DESCRIPTION
**Pull request by sgreensmith**
By design:
- when the form is submitted, the raw form data is sent to zendesk. Then:
- if successful, the user sees the confirmation page and the following happens in the background.
- temporary config files are created in the application's ./tmp directory
- the files are named  based on the department and service names provided in the
form
- the .yml files are then sent to zendesk and attached to the existing ticket as a new comment.
- if sending the config files generates any errors, these are hidden
from the user.
- Currently, only the <transaction>, <msa>, <experian-stub-idp>, and <post-office-stub-idp> files are created

Additional bug fixes:
- updated zendesk instructions link to point at alphagov
- moved the generation of user_account_attributes array into a library
- changed the default value of 'ShouldHubUseLegacySamlStandard' to false

Author: @sgreensmith

### CODE REVIEW

File: app/controllers/environment_config_form_controller.rb [hunk 1]
```diff
@@ -13,9 +13,24 @@ def submit
     end
 
     begin
-      @ticket = OnboardingFormService.new(ZendeskClientFactory.create).save @onboarding_form
+      onboarding_form_service = OnboardingFormService.new()
+      ticket_body = onboarding_form_service.generate_ticket_body(@onboarding_form)
+      @ticket = onboarding_form_service.save(ticket_body)
+
+      if !@ticket; raise 'problem creating ticket' end
```
Comment (vixus0) [inline app/controllers/environment_config_form_controller.rb]: Might read better as:

```ruby
raise 'problem creating ticket' if @ticket.nil?
```

File: app/controllers/environment_config_form_controller.rb [hunk 2]
```diff
@@ -13,9 +13,24 @@ def submit
     end
 
     begin
-      @ticket = OnboardingFormService.new(ZendeskClientFactory.create).save @onboarding_form
+      onboarding_form_service = OnboardingFormService.new()
```
Comment (vixus0) [inline app/controllers/environment_config_form_controller.rb]: Don't need the `()` on `.new`.

File: app/lib/transform_user_account_attributes.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class TransformUserAccountAttributes
+
+  def get_array(onboarding_form)
```
Comment (vixus0) [inline app/lib/transform_user_account_attributes.rb]: Move to form object?

File: spec/features/user_visits_form_spec.rb [hunk 4]
```diff
@@ -48,15 +49,17 @@
...
+    expect(page).to have_content('Your ticket has been created with the id #123456')
 
-  it 'should show confirmation page after a succesful form submit even if there\'s no ticket id' do
-    stub_request(:post, ZENDESK_TICKETS_URL).to_return(:status => 201, :body => {"ticket":{"subject":"Test","comment":{"value":"some comment"}}}.to_json, :headers => {})
-    submit_valid_form
-    expect(page).to have_content('Your ticket has been created with the id #unknown')
+    File.delete('./tmp/something-something.yml')
+    File.delete('./tmp/something-something-msa.yml')
+    File.delete('./tmp/experian-something-something.yml')
+    File.delete('./tmp/post-office-something-something.yml')
```
Comment (vixus0) [inline spec/features/user_visits_form_spec.rb]: Use stub IDP names.

File: app/controllers/environment_config_form_controller.rb (hunk 1 above)
Comment (sgreensmith) [reply app/controllers/environment_config_form_controller.rb]: changed

File: app/lib/transform_user_account_attributes.rb (hunk 3 above)
Comment (sgreensmith) [reply app/lib/transform_user_account_attributes.rb]: changed

File: spec/features/user_visits_form_spec.rb (hunk 4 above)
Comment (sgreensmith) [reply spec/features/user_visits_form_spec.rb]: @chrisholmes has said this is acceptable, the initial decision to not include any references to idp's referred to not sharing any styling or content that they owned.

**Comment by chrisholmes** [comment]
http://ruby-doc.org/stdlib-2.0.0/libdoc/tmpdir/rdoc/Dir.html and https://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html might help

**Pull request merged**